Automattic/wp-calypso
# Site Settings: Move General section to a separate chunk

We've been working a lot on the Site Settings recently. A lot of stuff has been introduced, enhanced or improved, but this also has some cons: the `settings` chunk is now 1.12MB 😱 

![](https://cldup.com/M1UdN8a_9R.png)

It is obvious that we need to split `settings` to smaller chunks. This has also been discussed in p4TIVU-5Zs-p2 and recommended by @dmsnell and @mtias.

So this PR moves the General settings section to a separate chunk. After this change, the chunk sizes are:

![](https://cldup.com/tEp1SfL0WM.png)
![](https://cldup.com/LxSSWOSNpO.png)

This is not introducing any functional or behavioral changes. But since it's affecting all of the routes and functionality that resides within the General settings section, extensive testing of the General section and its settings will be required.

To test:
* Checkout this branch, or get it going on calypso.live
* Go to `/settings/general/$site`, where `$site` is one of your WordPress.com sites.
* Verify the tab loads correctly with no errors/warnings.
* Change to another tab and come back to General, verify it works like before.
* Play with all settings in the General section and verify they are saved and retrieved correctly.
* Click on the "Delete Site" button and verify it works properly:
  * If you have a Free plan, it should direct you to `/settings/delete-site/$site`
  * If you have a paid plan, it should display a popover to cancel any upgrades first.
* Click on the "Start Over" button and verify it loads `/settings/start-over/$site` correctly.
* Click on the "Theme Setup" button and verify it loads `/settings/theme-setup/$site` correctly.
* If you're on the dev environment (`calypso.localhost`), click on the "Date and Time Format" button and verify it loads `/settings/date-time-format/$site` correctly.
* Try opening any of the above routes in a fresh page session and verify they all load correctly.
* Go to `/settings/general/$site`, where `$site` is one of your connected Jetpack sites.
* Verify the tab loads correctly with no errors/warnings.
* Change to another tab and come back to General, verify it works like before.
* Play with all settings in the General section and verify they are saved and retrieved correctly.
* If you're on the dev environment (`calypso.localhost`), click on the "Date and Time Format" button and verify it loads `/settings/date-time-format/$site` correctly.
* Don't hate me for the long test instructions list 😆
